### PR TITLE
feat(BetterSliding)!: 重构微调参数为 FineTuneQuantity / FineTuneFallback

### PR DESCRIPTION
**Pull request by single-ptilopsis**
## 背景

BetterSliding 在精确点击后原本只有两种收尾方式：默认继续用 `IncreaseButton` / `DecreaseButton` 逐步微调，或 `FinishAfterPreciseClick: true` 直接返回成功、完全不校验数量。实际调用场景需要更细的控制（差值不大才微调、不微调时朝某个方向补偿一次），而旧的微调重试路径还有一个会把失败上报为成功的缺陷。

本 PR 把「是否微调」重构为 `FineTuneQuantity` + `FineTuneFallback`，并修掉重试预算耗尽的判负问题。

### 1. 参数重构：`FinishAfterPreciseClick` → `FineTuneQuantity` + `FineTuneFallback`

| 参数 | 类型 | 默认 | 说明 |
| --- | --- | --- | --- |
| `FineTuneQuantity` | `bool` / `int` | `true` | `true` 始终微调；`false` 一律不微调；整数 `N >= 1` 仅当 `abs(current - target) <= N` 时才微调（允许「偏移进入阈值后切回微调」的混合行为） |
| `FineTuneFallback` | `string` | `"none"` | 仅在本次判定为「不微调」时生效：`none` 不做补偿直接收尾；`more` / `less` 在数量偏少 / 偏多时朝对应方向补偿并复查 |

- 两个参数都支持 `attach`（优先）与 `custom_action_param` 两条通道。
- `FinishAfterPreciseClick` 硬移除：类型、presence、归一化、attach 合并、日志与分支全部删除，不保留兼容代码、不做告警；外部仍传该字段时由 `encoding/json` 静默忽略，行为回落为默认的「继续微调」。
- `isSwipeOnlyMode` 纳入两个新参数的 presence，只传它们时不会被误判为仅滑动模式。

### 2. 不微调时的补偿：先经 `Reset2` 反向复位，再精确点击并复查

- 补偿方向沿 Start → End 正方向：`abs(dx) > abs(dy)` 取 x 轴，否则取 y 轴（平局取 y；Start/End 中心重合时取 y 轴与正方向并告警）。
- 第 k 次补偿的落点是基准坐标 `±k` 的累加值，不是在上一次坐标上继续叠加。
- 每次点击前先经新增节点 `BetterSlidingReset2` 把滑块滑到精确点击点的另一侧（避免上一次精确点击落在滑块本体上干扰下一次）：投影比例 `< 0.5`（靠近 Start）向 End 侧滑动，否则向 Start 侧滑动；恰好落在中线或轴跨度为 0 时取 Start 侧并告警；`end` 由 `Direction` 推导后整字段覆盖，复用既有 `buildSwipeEnd` / `buildResetSwipeEnd`。
- `FineTuneFallback: "none"`，或 `more` / `less` 但方向条件不成立时：复查一次后直接路由到 `BetterSlidingDone`，不再补偿。

### 3. 修复：微调重试预算耗尽被误判为成功

- 重试预算从 `BetterSlidingCheckQuantity` 的「识别心跳数」改挂到动作节点（`IncreaseQuantity` / `DecreaseQuantity` / `Reset2` 各 `max_hit: 4`），消除 off-by-one：每次点击 / 偏移的结果都会被下一次数量复查看到；`CheckQuantity` 自身不再设 `max_hit`。
- 删除空叶节点 `BetterSlidingFail`：该节点是默认 `DoNothing`，命中后会被框架判为成功，导致预算耗尽仍上报成功并写入结果节点。现在预算耗尽后 `CheckQuantity` 的 next 候选全部不可用，框架以 `Node.NextList.Failed` 判负，`runInternalPipeline` 返回 `false`，且不再覆盖 `TargetReachableOverrideEnable`。
- `BetterSlidingClearMaxHit` 的 `nodes` 列表补入三个预算节点，避免命中计数跨运行残留导致同一次外层任务内后续调用无法微调。
- 代价：预算耗尽后需要等框架 `timeout`（默认 20s）才报错。

### 4. 参数校验

- `FineTuneQuantity` 仅接受 bool 或整数；非整数与 `< 1` 报错。整数 `> maxFineTuneThreshold (math.MaxInt32)` 时饱和为 `enabled`（始终微调）——该阈值已远大于任何真实数量差值，语义等价，且不再执行 `int(v)`，避免旧实现钳制到 `float64(math.MaxInt)`（= 2^63）再转 `int` 回绕成 `math.MinInt`、把「超大阈值」反转成「从不微调」的问题。文档只约束 `N >= 1`。
- `FineTuneFallback` 大小写不敏感；空串与 JSON `null` 归一为 `none`；其他值报错。
- 显式 `null` 与键缺失统一按「未提供」处理（`hasNonNullRawKey`）：`FineTuneQuantity` 回落默认 `true`，`FineTuneFallback` 回落 `none`；显式 `null` 也不再意外把节点踢出 swipe-only。

### 5. Pipeline / 文档 / 测试

- `Main.json`：新增 `BetterSlidingReset2`；`CheckQuantity` 去掉 `max_hit` 并补入 `Reset2` 候选；`Increase` / `Decrease` 加 `max_hit: 4`；删除 `BetterSlidingFail` 及其引用。该文件大部分 diff 来自把 `{"type": ..., "param": ...}` 嵌套写法改为扁平写法，语义变更仅上述几条。
- `Test.json`：`__BS-5` 迁移到新参数（`FineTuneQuantity: false` + `FineTuneFallback: "none"`），新增 `__BS-7`（`more`）/ `__BS-8`（`less`）用例；`__BS-T-1` 补入两个用例，`__BS-T-4` / `__BS-T-5` 补 `next`；节点按名称排序，写法统一为扁平。
- 中英文 `docs/*/developers/components/better-sliding.md`：参数表 5 → 6，新增「不微调语义 / No-Fine-Tune Semantics」小节；澄清 `TargetReachableOverrideEnable` 只表示解析后的目标可达，与最终调整结果无关。
- 过程设计文档 `.dev_doc/*` 已在分支内删除，结论由代码注释、文档与提交信息承载。

## 兼容性

- 破坏性：`FinishAfterPreciseClick` 移除，外部若依赖该参数请改用 `FineTuneQuantity: false`（+ 可选的 `FineTuneFallback`）。
- `BetterSlidingFail` 节点删除：仓库内无引用；外部任务若引用过该节点名需移除。
- 预算耗尽的行为由「静默成功」变为「失败」，且失败前需等框架 `timeout`（默认 20s）。

## Summary by CodeRabbit

- **新功能**
  - Better Sliding 支持数量微调，可直接结束，或按“更多/更少”方向补偿后复查。
  - 新增复位与再次检查流程，提升精确点击后的处理能力。

- **变更**
  - `FineTuneQuantity` 与 `FineTuneFallback` 替代原有完成后行为配置。
  - 更新滑动流程节点及测试配置，统一动作参数格式。
  - 超过微调阈值时改为警告并按始终微调处理。

- **文档**
  - 补充微调参数、补偿方向及目标可达状态说明。

### CODE REVIEW

File: agent/go-service/bettersliding/handlers.go [hunk 1]
```diff
@@ -601,6 +600,256 @@ func (a *BetterSlidingAction) handleDone(_ *maa.Context, _ *maa.CustomActionArg)
...
+		nodeBetterSlidingDone,
+		buttonTarget{},
+		0,
+	); err != nil {
+		if errors.Is(err, errCheckQuantityBranchNextOverride) {
+			logEvent.Err(err).Msg("failed to override next to done without fine-tuning")
+		} else {
+			logEvent.Err(err).Msg("failed to override done node without fine-tuning")
+		}
+		return false
```
[the remark this answers is not in the document]
Comment (single-ptilopsis) [reply agent/go-service/bettersliding/handlers.go]: 已修复：`cf0af6ce`。

确认该问题属实，并已用仓库固定的 `zerolog v1.34.0` 实测复现——`a.logger.Info()` 返回的 Event 级别在创建时即固化，失败分支复用同一 `logEvent` 调 `Err(err)` 只会追加 `error` 字段，输出仍是 `INF`。

影响比「级别错标」更大：`agent/go-service/logger.go:43-49` 的 console writer 设了 `minLevel: zerolog.ErrorLevel`，所以这条失败不只被记为 `INF`，还**完全不会出现在控制台**，只留在 `debug/go-service.log` 里，错误级告警会漏掉。

改法与你的建议一致：失败分支内新建 `a.logger.Error().Err(err)`，成功路径另建 Info Event。两条分支消息与全部结构化字段（`fine_tune_fallback`/`axis`/`end_sign`/`step_sign`/`current_quantity`/`target_quantity`）逐字保留。

另已全仓扫描确认：本处是唯一「非 Error 级 Event 变量后续调用 `.Err()`」的站点，无同类缺陷需一并修。

File: docs/en_us/developers/components/better-sliding.md [hunk 2]
```diff
@@ -53,14 +53,15 @@ Suitable for scenarios where you want to slide to the maximum/minimum. Parameter
...
 | `TargetQuantity` | `int` (positive integer) | Yes | Target quantity. The desired final slider value, which must be greater than 0. |
 | `TargetQuantityType` | `string` | No | How to interpret `TargetQuantity`. `"Value"` (default): absolute count; `"Percentage"`: percentage of `availableQuantity` (1–100), rounded and clamped. |
 | `ReverseTarget` | `bool` | No | When `true`, resolves the target from the available quantity: Value mode uses `availableQuantity - TargetQuantity`; Percentage mode uses the remaining percentage. Default `false`. |
-| `FinishAfterPreciseClick` | `bool` | No | When `true`, returns success immediately after a precise click, without entering the quantity validation and fine-tuning process. Default `false`. |
+| `FineTuneQuantity` | `bool` or `int` | No | Whether to keep fine-tuning via Increase/Decrease after the precise click. `true` (default): always fine-tune; `false`: never fine-tune; integer `N` (must be `>= 1`): fine-tune only when `abs(current - target) <= N`. |
```
[the remark this answers is not in the document]
Comment (single-ptilopsis) [reply docs/en_us/developers/components/better-sliding.md:63]: 问题属实，但按维护者决定，采用**改代码行为**而非改文档的方式解决，已随 `b95c7684` 推送。

你指出的不一致确实存在：`newThresholdFineTuneQuantity` 拒绝 `> math.MaxInt32`，而文档只写 `N >= 1`，`2147483648` 符合文档却会让参数校验失败。

但我们没有给文档补上限，而是把超界整数改为**饱和为 `enabled`（始终微调）**：

```go
if v > maxFineTuneThreshold {
    log.Warn().
        Float64("fine_tune_quantity", v).
        Int("max_fine_tune_threshold", maxFineTuneThreshold).
        Msg("FineTuneQuantity threshold above max, treated as always fine-tune")
    return fineTuneQuantity{enabled: true}, nil
}
```

理由是该阈值已远大于任何真实数量差值，语义上与「始终微调」等价，硬报错反而给用户增加了一个无实际意义的上限约束。这样文档的 `N >= 1` 就是完备描述，两处 `better-sliding.md` 无需改动（也因此不会新增第三条待同步的文案）。

顺带修掉了溢出隐患：新分支不再执行 `int(v)`。`N = MaxInt32`（`thresholdMode`）与 `N = MaxInt32 + 1`（`enabled`）对任意真实数量差值结果一致，边界无跳变；`v < 1` 与非整数值仍照旧报错。`normalize.go` 与 `types.go` 的注释已同步改述。

**Comment by single-ptilopsis** [comment]
@coderabbitai 已按复核结论处理本轮的 2 条 inline comment 与 2 项 pre-merge check，改动见 `cf0af6ce`、`b95c7684`。

## 已修复

- **`handlers.go` 失败分支日志级别**（inline，属实）：确认为真实缺陷，control console 因 `minLevel: ErrorLevel` 会完全漏掉该失败。已改为在失败分支新建 `a.logger.Error().Err(err)`。
- **`FineTuneQuantity` 文档边界**（inline，属实但换了解法）：未给文档补 `1 <= N <= 2147483647`，而是把超界整数改为饱和为 `enabled`（始终微调）。该阈值已远大于任何真实数量差值，等价于「始终微调」，硬报错只是给用户增加无实际意义的上限约束。这样 `N >= 1` 就是完备描述，两处 `better-sliding.md` 无需改动。同时不再执行 `int(v)`，消除溢出隐患。

## 明确不采纳（两项 pre-merge check）

**1. Pipeline Standards Compliance —— 不采纳**

你的事实描述经逐条核对**全部准确**：`BetterSlidingReset2` 确为本 PR 新增，确实是「识别 `BetterSlidingSwipeButton` → `Swipe` → 无条件 `next` 到无识别的 `BetterSlidingPreciseClick` → `Click`」，确实缺少一次状态验证，base 的 `BetterSlidingCheckQuantity` 也确实没有该路由。

但「本 PR 新增了违反规范的结构」这一定性不成立，因此不插识别节点：

1. base `origin/v2` 中 `BetterSlidingFindEnd`（识别 `SwipeButton`）→ `BetterSlidingPreciseClick`（`DirectHit` Click）已是完全相同的模式；base 的 `handlePreciseClick` 也早在把 `BetterSlidingReset`（DirectHit Swipe）的 `next` override 成 `PreciseClick`（base `handlers.go:477`）。本 PR 只是把这条既有路径改用静态 `next` 表达。
2. 全仓扫描「Swipe → 无识别节点」的边共 6 条，另有 4 条在 `BatchAddFriends` / `SyncItemData` / `SyncValuablesItemData` / `ResourceRecycleCollect` 等与本 PR 无关的既有 Pipeline 中。
3. `Reset2` 带 `max_hit: 4`，且 `Click` 后立即经 `JumpBackNode` 回到 `CheckQuantity` 重新识别，并非「永不复查」。

即真正根因是既有设计模式而非本 PR 引入。该场景已确认稳定，作为独立 issue 统一评估更合适，不宜单独阻塞本 PR。

**2. Docstring Coverage 60% < 80% —— 不采纳**

80% 是 CodeRabbit 的组织级阈值，不是本仓库的规范。`.agents/skills/go-service-guide/SKILL.md:114-117` 要求的是**导出符号**必须有注释；被点出的 `nudgedClickTarget`、`absInt`、`signInt` 均为未导出函数，而 `absInt` / `signInt` 是两行平凡实现，补注释只会变成冗余（参见本仓库 `dc991f54 style(BetterSliding): 精简冗余注释`）。本轮只改述既有注释文字，未新增注释块。
